makefile

```diff
@@ -1,61 +0,0 @@
-# Binaries
-UGLIFY = ./node_modules/.bin/uglifyjs
-BROWSERIFY = ./node_modules/.bin/browserify
-KARMA = ./node_modules/karma/bin/karma
-JSHINT = ./node_modules/.bin/jshint
-DEREQUIRE = ./node_modules/.bin/derequire
-
-# Build debug and minified libraries
-all: \
-	dist/tangram.min.js \
-	dist/tangram.debug.js
-
-debug: dist/tangram.debug.js
-
-dist/tangram.debug.js: .npm $(shell ./build_deps.sh)
-	node build.js --debug=true --require './src/module.js' --runtime | $(DEREQUIRE) > dist/tangram.debug.js
-
-dist/tangram.min.js: dist/tangram.debug.js
-	$(UGLIFY) dist/tangram.debug.js -c warnings=false -m -o dist/tangram.min.js
-	@gzip dist/tangram.min.js -c | wc -c | awk '{ printf "%.0fk minified+gzipped\n", $$1 / 1024 }'
```

Comment:
Derequire is used to uglify the original require() calls, which remain in
the bundled package. Without it, other browserify-based applications that
require('tangram') will attempt to parse Tangram's require() calls, but
fail because Tangram's dependencies are not present in the parent
application's dependency list.

On Fri, Apr 15, 2016, 11:50 AM Brett Camper <notifications@github.com>
wrote:

> In makefile
> <https://github.com/tangrams/tangram/pull/291#discussion_r59894392>:
>
> > -JSHINT = ./node_modules/.bin/jshint
> > -DEREQUIRE = ./node_modules/.bin/derequire
> > -
> > -# Build debug and minified libraries
> > -all: \
> > -	dist/tangram.min.js \
> > -	dist/tangram.debug.js
> > -
> > -debug: dist/tangram.debug.js
> > -
> > -dist/tangram.debug.js: .npm $(shell ./build_deps.sh)
> > -	node build.js --debug=true --require './src/module.js' --runtime | $(DEREQUIRE) > dist/tangram.debug.js
> > -
> > -dist/tangram.min.js: dist/tangram.debug.js
> > -	$(UGLIFY) dist/tangram.debug.js -c warnings=false -m -o dist/tangram.min.js
> > -	@gzip dist/tangram.min.js -c | wc -c | awk '{ printf "%.0fk minified+gzipped\n", $$1 / 1024 }'
>
> Spitting out the gzipped size has kept me conscious of bundle size (vs.
> proactively checking it periodically), I think it's a nice thing to keep.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/tangrams/tangram/pull/291/files/790ac72f2ee2cbfab7aaa2312a172eae4343e7fb..db1c1208bf1ff387fb5569e55400f13f51cc0afa#r59894392>
>